rfcs/20200928-keras-cv-single-stage-2d-object-detection.md

```diff
@@ -0,0 +1,431 @@
+# keras-cv Single Stage Two-Dimensional Object Detection API
+
+| Status        | Proposed      |
+:-------------- |:---------------------------------------------------- |
+| **Author(s)** | Zhenyu Tan (tanzheny@google.com), Francois Chollet (fchollet@google.com)|
+| **Contributor(s)** | Pengchong Jin (pengchong@google.com)|
+| **Updated**   | 2020-09-28                                           |
+
+## Objective
+
+We aim at providing the core primitive components for training and serving single-stage two-dimensional object
+detection models, such as Single-Shot MultiBox Detector (SSD), RetinaNet, and You-Only-Look-Once (YOLO).
+
+## Key Benefits
+
+Single-stage object detection models are a state-of-art technique that powers many computer vision tasks, they provide
+faster detection compared to two-stage models (such as FasterRCNN), while maintaining comparable performance.
+
+With this proposal, Keras users will be able to build end-to-end models with a simple API.
+
+## Design overview
+
+This proposal includes the specific core components for building single-stage object detection models. It does not,
+include, however, include:
+
+1. Data augmentation, such as image and groundtruth box preprocessing
+2. Model backbone, such as DarkNet, or functions to generate feature maps
+3. Detection heads, such as Feature Pyramid
+4. metrics utilities such as COCO Evaluator, or visualization utils.
+
+Data augmentation will be included as a separate RFC that handles a
+broader context than object detection.
+
+Model backbone and detection heads are model-specific, we anticipate them to be analyzed and proposed in 
+`keras.applications` for heavily used patterns, however the user can build them easily using Keras.
+
+#### Training
+
+Case where a user want to train from scratch:
```

Comment:
Would there be the option to train from a pre-defined checkpoint? For example, start from
a model pre-trained on coco data set and then do transfer learning. Having those weights
from coco for example for the different backbones in `tf.keras.applications` would be nice.
Much like we have the imagenet weights and there is the option to remove the head and have the pre-trained
feature extractor (as it is today for classification for example in `tf.keras.applications`.

Reply:
Yes! I forgot to mention this. Updated.